addons/portal/static/src/js/portal_chatter.js

```diff
@@ -91,24 +94,54 @@ var PortalChatter = publicWidget.Widget.extend({
             route: '/mail/chatter_fetch',
             params: self._messageFetchPrepareParams(),
         }).then(function (result) {
-            self.set('messages', self.preprocessMessages(result['messages']));
+            self.set('grouped_messages', self.preprocessMessages(result['messages']));
             self.set('message_count', result['message_count']);
+            self.set('attachment_ids', result['attachment_ids']);
             return result;
         });
     },
+
     /**
-     * Update the messages format
+     * Update the messages format and store messages as a list
      *
      * @param {Array<Object>} messages
      * @returns {Array}
      */
-    preprocessMessages(messages) {
-        _.each(messages, function (m) {
-            m['author_avatar_url'] = _.str.sprintf('/web/image/%s/%s/author_avatar/50x50', 'mail.message', m.id);
-            m['published_date_str'] = _.str.sprintf(_t('Published on %s'), moment(time.str_to_datetime(m.date)).format('MMMM Do YYYY, h:mm:ss a'));
-            m['body'] = Markup(m.body);
+    preprocessMessages: function (messages) {
+        var groupedMessages = {};
+        var groupIndex = 0;
+        const DATE_FORMAT = time.getLangDateFormat();
+        const today = moment().format(DATE_FORMAT);
+        const self = this;
+
+        messages.forEach(function(message, messageIndex) {
+            self._preprocessMessage(message, messageIndex);
+            var groupDate = message.published_date.format(DATE_FORMAT);
+            if (groupDate in groupedMessages) {
+                groupedMessages[groupDate].messages.push(message);
+            } else {
+                groupedMessages[groupDate] = {
+                    'groupIndex': groupIndex++,
```

Comment:
Not understanding this indexes management .... seems a bit yolo ?

Reply:
It is used only to sort the list, I will rename it sequence